coordinatorlib/src/main/scala/com/socrata/datacoordinator/secondary/messaging/eurybates/EurybatesProducer.scala

```diff
@@ -0,0 +1,74 @@
+package com.socrata.datacoordinator.secondary.messaging.eurybates
+
+import java.util.{Properties, UUID}
+import java.util.concurrent.Executor
+
+import com.rojoma.json.v3.codec.JsonEncode
+import com.socrata.datacoordinator.secondary.messaging.{Message, NoOpProducer, Producer}
+import com.socrata.eurybates
+import com.socrata.eurybates.zookeeper.ServiceConfiguration
+import com.socrata.thirdparty.typesafeconfig.ConfigClass
+import com.socrata.zookeeper.ZooKeeperProvider
+import com.typesafe.config.Config
+import org.slf4j.LoggerFactory
+
+class EurybatesProducer(sourceId: String,
+                        zkp: ZooKeeperProvider,
+                        executor: Executor,
+                        properties: Properties) extends Producer {
+  val log = LoggerFactory.getLogger(classOf[Producer])
+
+  private val producer = eurybates.Producer(sourceId, properties)
+  private val serviceConfiguration = new ServiceConfiguration(zkp, executor, setServiceNames)
+
+  def start(): Unit = {
+    producer.start()
+    serviceConfiguration.start()
+  }
+
+  def setServiceNames(serviceNames: Set[String]): Unit = {
+    producer.setServiceNames(serviceNames)
+  }
+
+  def shutdown(): Unit = {
+    producer.stop()
+  }
+
+  def send(message: Message): Unit = {
+    log.debug("Sending message: {}", message)
+    try {
+      producer.send(eurybates.Message(EurybatesMessage.tag(message), JsonEncode.toJValue(message)))
+    } catch {
+      case e: IllegalStateException =>
+        log.error("Failed to send message! Are there no open sessions to queues?", e)
+    }
+  }
+}
+
+class ProducerConfig(config: Config, root: String) extends ConfigClass(config, root) {
+  def p(path: String) = root + "." + path
```

Comment:
May be you can even do something like val core = new CoreConfig(getRawConfig("core"))